source/brailleDisplayDrivers/albatross/constants.py

```diff
@@ -0,0 +1,306 @@
+"""
+
+READ_TIMEOUT = 0.2
+WRITE_TIMEOUT = 0
+SLEEP_TIMEOUT = 0.2
+"""How long to sleep between port init or open retries."""
+
+MAX_INIT_RETRIES = 20
+"""
+How many times port initialization or opening is tried.
+This large value is required for braille to start.
+This should not effect other devices with the same PID and VID,
+because this device is detected as last one.
+"""
+
+RESET_COUNT = 10
+"""
+How many times to try to reset I/O buffers.
+Possibly due to write operations of other drivers with same PID&VID,
+device appears to send init packets it should not.
+Although reset is not ideal, unexpected packets are discarded.
+In addition, there are arbitrary number of redundant normal init packets
+which can be safely ignored.
+"""
+
+RESET_SLEEP = 0.05
+"""How long to sleep between I/O buffer resets."""
+
+WRITE_QUEUE_LENGTH = 20
+
+
+class Keys(IntEnum):
+	"""Defines key names and values.
+	For routing keys see L{RoutingKeyRange}.
+"""
+	attribute1 = 1
+	attribute2 = 42
+	f1 = 83
+	f2 = 84
+	f3 = 85
+	f4 = 86
+	f5 = 87
+	f6 = 88
+	f7 = 89
+	f8 = 90
+	home1 = 91
+	end1 = 92
+	eCursor1 = 93
+	cursor1 = 94
+	up1 = 95
+	down1 = 96
+	left = 97
+	up2 = 98
+	lWheelRight = 103
+	lWheelLeft = 104
+	lWheelUp = 105
+	lWheelDown = 106
+	attribute3 = 151
+	attribute4 = 192
+	f9 = 193
+	f10 = 194
+	f11 = 195
+	f12 = 196
+	f13 = 197
+	f14 = 198
+	f15 = 199
+	f16 = 200
+	home2 = 201
+	end2 = 202
+	eCursor2 = 203
+	cursor2 = 204
+	up3 = 205
+	down2 = 206
+	right = 207
+	down3 = 208
+	rWheelRight = 213
+	rWheelLeft = 214
+	rWheelUp = 215
+	rWheelDown = 216
+
+
+MAX_COMBINATION_KEYS = 4
+"""Maximum number of keys in key combination."""
+
+CONTROL_KEY_CODES: Set[Keys] = {
+	Keys.attribute1,
+	Keys.attribute2,
+	Keys.f1,
+	Keys.f2,
+	Keys.f7,
+	Keys.f8,
+	Keys.home1,
+	Keys.end1,
+	Keys.eCursor1,
+	Keys.cursor1,
+	Keys.attribute3,
+	Keys.attribute4,
+	Keys.f9,
+	Keys.f10,
+	Keys.f15,
+	Keys.f16,
+	Keys.home2,
+	Keys.end2,
+	Keys.eCursor2,
+	Keys.cursor2,
+}
+"""Ctrl keys which may start key combination."""
+
+LEFT_RIGHT_KEY_CODES: Dict[Keys, Keys] = {
+	Keys.attribute1: Keys.attribute3,
+	Keys.attribute2: Keys.attribute4,
+	Keys.f1: Keys.f9,
+	Keys.f2: Keys.f10,
+	Keys.f3: Keys.f11,
+	Keys.f4: Keys.f12,
+	Keys.f5: Keys.f13,
+	Keys.f6: Keys.f14,
+	Keys.f7: Keys.f15,
+	Keys.f8: Keys.f16,
+	Keys.home1: Keys.home2,
+	Keys.end1: Keys.end2,
+	Keys.eCursor1: Keys.eCursor2,
+	Keys.cursor1: Keys.cursor2,
+	Keys.up1: Keys.up3,
+	Keys.down1: Keys.down2,
+	Keys.lWheelRight: Keys.rWheelRight,
+	Keys.lWheelLeft: Keys.rWheelLeft,
+	Keys.lWheelUp: Keys.rWheelUp,
+	Keys.lWheelDown: Keys.rWheelDown,
+}
+"""Connects corresponding left and right side keys.
+Dictionary keys are left side keys values, and dictionary values are
+corresponding right side keys values. Used with custom key layouts.
+"""
+
+KEY_LAYOUT_MASK = 5
+"""Used to extract bits 1 - 3 from settings byte to determine key layout.
+See L{KeyLayouts}.
+"""
+
+
+class KeyLayouts(IntEnum):

```

Comment:
This should be singular

```suggestion
class KeyLayout(IntEnum):
```